### ISSUE TITLE
[Question] how to use "weight" barcode ?

### ISSUE THREAD

**Issue by truchosky** [opensourcepos/opensourcepos#2135]
Hi guys,

I read this #1243, however I couldn't find any example or help about the use, it needs a special config?, I have the weight and it prints a barcode but I dont know how to config ospos to read it

**Comment by WebShells** [comment]
Please fill in template issue so we can help you...
Thanks

**Comment by daN4cat** [comment]
@WebShells you should know how it works based on #1243

**Comment by WebShells** [comment]
@daN4cat for sure, just making sure every user fills in issue template for further requests...
@truchosky please share with me barcode format you are using, ex: department / lfcode / quantity? How many charcs each?

**Comment by WebShells** [comment]
@truchosky ? If you can share the format so we be able to help?

**Comment by WebShells** [comment]
A Wiki Page was added in order to explain the requested...  

[Weighted Barcodes](https://github.com/opensourcepos/opensourcepos/wiki/Weighted-Barcodes)

If you still need help reopen this issue...

**Comment by WebShells** [comment]
@jekkos yes I already linked it from the main wiki page and added it into the sidebar... I guess with what i added it should somehow be clear for users how to use it.

**Comment by truchosky** [comment]
thx you !

**Comment by izeblood** [comment]
> department
Where did this department coming from since its only Id | Barcode | Item Name | Category | Company Name | Wholesale Price | Retail Price | Quantity being seen in OSPOS item module?

**Comment by akhilnasi** [comment]
Hi ... I need the format for weigh scale barcode ...(Department,Quatity,Price)...I used the format which you have mentioned but i didnot ..do anyone can help me..Sorry if this topic is closed am sorry ...Because i needed it urgently...
My format:   2 char for department and 6 qty and 5 price

**Comment by izeblood** [comment]
02 / \d {5} / \w {6} : 02(\d{5})(\w{6}) 
this will solve your problem....

**Comment by izeblood** [comment]
How to implement this: for example: in a meatshop, you have to make item list first with price per kilo.
then by the use of your digital weighing scale that prints price tags... it embed price x quantity = the amount to sell.

**Comment by akhilnasi** [comment]
@izeblood ...
for example:  consider Fish
Fish price always changes on daily basis and there are lot of fishes that a shopkeeper buys.
so they cannot change the price daily of each fish in item module.
instead of that 
i wanted to do like this
02->department
05->item
06->price

this i wanted.
Thank you

**Comment by truchosky** [comment]
hi guys, like @izeblood asked, how do you work with kg?

**Comment by jekkos** [comment]
You can use decimal precision if I'm not mistaken. Or just adapt the code, whatever works for you.

**Comment by truchosky** [comment]
I noticed 02 "department" make the change for it self, if you set 02(\d{5})(\w{6}) the w takes the number and pass it to kg, worked flawless for me

**Comment by crblack76** [comment]
Dear colleagues, I have loaded the variable
02 (\ d {5}) (\ w {6}) according to wiki in the ospos version that I have running 3.3.1
and so far I couldn't get it to work.
I have my doubts about it, I give you an example.
this is the encoded data
corresponding to my scale label.
20 (XXXXX) (XXXXXX)
Department - Item - Weight
Example:
20 00001 002551
![label](https://user-images.githubusercontent.com/73746212/97783211-fc6b9f80-1b74-11eb-8467-d782e3dc2bb8.jpg)
Since my scale does not allow to change the order of the sequence, what I did was modify the variable in this way.
02 (\ w {5}) (\ d {6}) with no result.
I then encoded the data in the correct order with an online barcode generator and it didn't work for me either.
So I am wondering what is the correct way to configure this feature.

On my scale, the department used for products that are weighed is two digits "20" and for products that are not weighed "24".
In another system that was used when uploading an item, we included the department along with the item ID, which in the example would be (20) 00001.
Any suggestions where it could be going wrong?
Thank you
